### PR TITLE
ext2: adapt atasrv interface changes

### PR DESCRIPTION
**Pull request by Maxez**

### CODE REVIEW

File: ext2/atasrv.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#include <sys/types.h>
+
+
+extern ssize_t atasrv_read(id_t *id, offs_t offs, char *buff, size_t len);
+
+
+extern ssize_t atasrv_write(id_t *id, offs_t offs, const char *buff, size_t len);
+
+
+#endif
```
Comment (agkaminski) [inline ext2/atasrv.h:28]: This is API of atasrv from the phoenix-rtos-devices? Please take a look for example on stm32-multi headers installing. Copying headers makes further development error prone.

File: ext2/sb.c [hunk 2]
```diff
@@ -58,27 +56,23 @@ int ext2_read_sb(id_t *devId, ext2_fs_info_t *f)
...
 void ext2_init_fs(id_t *devId, ext2_fs_info_t *f)
 {
-	int size, err;
+	int size;
 	void *buff;
 
 	f->block_size = EXT2_SB_SZ << f->sb->log_block_size;
 	f->blocks_count = f->sb->blocks_count;
-	printf("ext2: Mounting %.2f MiB partition\n", (float)(f->block_size * f->blocks_count)/1024/1024);
+	printf("ext2: Mounting %u B partition\n", f->block_size * f->blocks_count);
```
Comment (agkaminski) [inline ext2/sb.c:75]: It will overflow on partitions bigger than 4 GiB.

File: ext2/atasrv.h (hunk 1 above)
Comment (Maxez) [reply ext2/atasrv.h]: I got rid of the atasrv.h header and used read and write callbacks passed in to libext2_mount() (mounting is de facto filesystem initialization) for device access instead.

File: ext2/sb.c (hunk 2 above)
Comment (Maxez) [reply ext2/sb.c]: Fixed by using PRIu64 for printing uint64_t values.

File: ext2/ext2.h [hunk 3]
```diff
@@ -240,6 +241,11 @@ typedef struct {
 		ext2_object_t 	*lru;
 } ext2_fs_objects_t;
 
+
+typedef ssize_t (*read_callback)(id_t *, offs_t, char *, size_t);
+typedef ssize_t (*write_callback)(id_t *, offs_t, const char *, size_t);
```
Comment (agkaminski) [inline ext2/ext2.h:246]: These definitions are already in libext2.h

**Pull request merged**